proposals/vector_to_random_points.json

```diff
@@ -0,0 +1,85 @@
+{
+    "id": "vector_to_random_points",
+    "summary": "Sample random points from geometries",
+    "description": "Generate a vector data cube of points by sampling random points from input geometries.\n\nThrows a `CountsMissing` exception if `geometry_count` and `total_count` are both unrestricted (i.e. set to `null`).",
```

Comment:
How would you express that schematically while not restricting current functionality?

Reply:
I don't completely understand what my proposal would restrict.

now:
- default:  `geom_count=1` and `total_count=null`  -> 1 sample per geom
- `geom_count=null`   and `total_count=null`   ->  invalid

my proposal would be to change that invalid case to
- `geom_count=null`   and `total_count=null`   ->  total_count=number of geoms so basically: 1 sample per geom


advantage is that you then can make `geom_count=null` by default and the user doesn't have to explicitly set geom_count to null if they just want to set total_count

Reply:
so we would have the following scenarios covered with @soxofaan proposal:

- `geom_count=null` and `total_count=null` -> return 1 point each geometry
- `geom_count=null` and `total_count=100` -> return 100 points, distributed between the available geometries
- `geom_count=10` and `total_count=null` -> return 10 points each geometry
- `geom_count=10` and `total_count=1`-> return CountMismatch Exception or return 10 points and ignore total_count?

Reply:
yes indeed, with defaults `geom_count=null` and `total_count=null` , which give same result as current defaults, but usage is simpler

To iterate on your examples:
- `vector_to_points()`  -> 1 point per geometry
- `vector_to_points(total_count=100)` -> 100 points, distributed between the available geometries   or `CountMismatch` if there are more than 100 geometries
- `vector_to_points(geom_count=10)` -> 10 points per geometry
- `vector_to_points(geom_count=10, total_count=100)` -> at most 10 points per geometry and at most 100 points in total:
    - 10 geometries or less:  -> 10 points per geometry
    - between 10 and 100 geometries: -> 100 samples distributed between these geometries (with at least one per geometry
    - more than 100 geometries: `CountMismatch`
- `vector_to_points(geom_count=10, total_count=1)` : a bit of strange edge case, but it's still possible to honor the maximum constraints:
    - 1 geometry ->   1 point per geometry
    - 2 or more geometries:  `CountMismatch`

Reply:
Sorry, but I think there's an issue here: So if total_count = null means number of geometries, then `vector_to_points(geom_count=10)` for e.g. 10 geometries would return 1 point per geometry as implicitly the total_count is 10.

Reply:
I assume that total_count always trumps geom_count (which we should probably document).

Reply:
> Sorry, but I think there's an issue here: ... So you'd always need to specify a total_count now which is number of geom * geom_count. 

Indeed, good point. Explaining the my intended behavior  through a "dynamic" default value of  `total_count` is probably not the best solution.  


Taking a step back, I think there are two concerns I want to tackle simultaneously:
- giving  `geom_count=null` and `total_count=null` currently causes `CountsMissing` error. I think we should provide a non-error behavior (e.g. 1 sample per geometry)
- if the user just want to specify `total_count=100`, they should not forget to also explicitly set `geom_count` to `null` otherwise it won't work and they just get 1 sample per geometry (because that is the default for `geom_count`). I don't like this UI/UX-wise.

My proposal to tackle both is:
-  give `geom_count` default value `null`
- allow `geom_count=null` + `total_count=null` : 1 sample per geometry

Achieving the latter can be done (instead of the "dynamic" `total_count` default value trick I suggested earlier) by replacing the note about `CountsMissing` with something like:

> If `geometry_count` and `total_count` are both unrestricted (i.e. set to `null`), one sample per geometry is used.

Reply:
> I assume that total_count always trumps geom_count (which we should probably document).

I don't understand what you mean here. They are both currently described as "maximum", which implies that the most tight constraint trumps the other by definition

Reply:
Another question: how would you distribute the points among the geometries if `total_count = 100` and `geom_count = null` ?

My idea was to compute an estimate of the total area of all the geometries, compute the density total_area / total_count and then compute how many samples each geometry based on their area.
Too complex?

Reply:
That question is even more general: 
how would you distribute the points among the geometries if `geom_count > total_count /  number_of_geometries` or `geom_count=null` ?

Reply:
Opened issues https://github.com/Open-EO/openeo-processes/issues/344 and https://github.com/Open-EO/openeo-processes/issues/345 to keep track of these questions.